sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/DistinctMapTest.java

```diff
@@ -0,0 +1,109 @@
+    }
+
+    @Test(groups = "unit", dataProvider = "distinctMapArgProvider")
+    public void objectValue(DistinctQueryType queryType) {
+        String resource = String.format("{ "
+                                            + "\"id\": \"%s\", "
+                                            + "\"mypk\": \"%s\", "
+                                            + "\"sgmts\": [[6519456, 1471916863], [2498434, 1455671440]]"
+                                            + "}", 117546, "xxyyzz-abc");
+        DistinctMap distinctMap = DistinctMap.create(queryType, null);
+
+        Utils.ValueHolder<String> outHash = new Utils.ValueHolder<>();
+        boolean add = distinctMap.add(resource, outHash);
+        assertThat(add).as("Value should be added first time").isTrue();
+        boolean add2 = distinctMap.add(resource, outHash);
+        assertThat(add2).as("same value should not be added again").isFalse();
+
+        resource = String.format("{ "
+                                     + "\"id\": \"%s\", "
+                                     + "\"mypk\": \"%s\", "
+                                     + "\"sgmts\": [[6519456, 1471916863], [2498434, 1455671440]]"
+                                     + "}", 117546, "xxy%zz-abc");
+        boolean add3 = distinctMap.add(resource, outHash);
+        assertThat(add3).as("different value should be added again").isTrue();
+
+    }
+
+    @Test(groups = "unit", dataProvider = "distinctMapArgProvider")
+    public void nullValue(DistinctQueryType queryType) {
+        DistinctMap distinctMap = DistinctMap.create(queryType, null);
+        Utils.ValueHolder<String> outHash = new Utils.ValueHolder<>();
+        boolean add = distinctMap.add(null, outHash);
+        assertThat(add).as("Value should be added first time").isTrue();
+        boolean add2 = distinctMap.add(null, outHash);
+        assertThat(add2).as("same value should not be added again").isFalse();
+    }
+
+    @Test(groups = "unit", dataProvider = "distinctMapArgProvider")
+    public void objectOrder(DistinctQueryType queryType) {
+        String resource1 = String.format("{ "
+                                            + "\"id\": \"12345\", "
+                                            + "\"mypk\": \"abcde\""
+                                            + "} ");
+
+        String resource2 = String.format("{ "
+                                             + "\"mypk\": \"abcde\","
+                                             + "\"id\": \"12345\""
+                                             + "} ");
+
+        DistinctMap distinctMap = DistinctMap.create(queryType, null);
+
+        Utils.ValueHolder<String> outHash = new Utils.ValueHolder<>();
+        boolean add = distinctMap.add(resource1, outHash);
+        assertThat(add).as("Value should be added first time").isTrue();
+        boolean add2 = distinctMap.add(resource2, outHash);
+        assertThat(add2).as("Order of objects in map should be treated same").isTrue();
```

Comment:
This should be `false` right? since the ordering does not change the hash.

Reply:
Yeah it should be, corrected